### PR TITLE
Create separate site palettes. #99.

### PR DESCRIPTION
**Pull request by frano-m**
### Reviewers

#99 

@MillenniumFalconMechanic 

### Changes
- added custom palette to each site config.
- refactored palette to mirror Mui ThemeOptions palette shape.
  - refactored any use of palette colors to handle changed palette shape.
  - added temporary fix to `Text` component to handle changed palette shape.
- refactored `Socials` to take styled IconButton for header and footer - unique custom palette required - outside of normal palette shape.
  - resolved footer hover color.

### CODE REVIEW

File: explorer/app/components/FileCounts/Item.tsx [hunk 1]
```diff
@@ -13,10 +14,10 @@ export const Item = ({ count, name }: ItemProps): JSX.Element => {
       <Text variant="text-body-400-2lines" customColor="ink">
         {name}
       </Text>
-      <Text
+      <Typography
         variant="text-body-400-2lines"
-        customColor="primaryAnvil"
-      >{`${count} file(s)`}</Text>
+        color="primary"
```
Comment (frano-m) [inline explorer/app/components/FileCounts/Item.tsx:19]: Reverted to `primary` - site configured palette will set colour.

Comment (frano-m) [reply explorer/app/components/FileCounts/Item.tsx:19]: No need for `Text` component.

File: explorer/app/components/Layout/components/Footer/footer.tsx [hunk 2]
```diff
@@ -40,7 +43,12 @@ export const Footer = ({ footer }: Props): JSX.Element => {
             </NLink>
           ))}
           {feedbackForm && <>{/* TODO feedback form */}</>}
-          <Socials buttonSize="small" socials={socials} sx={{ gap: 2 }} />
+          <Socials
+            buttonSize="small"
+            IconButtonElType={IconButtonSocialsFooter}
```
Comment (frano-m) [inline explorer/app/components/Layout/components/Footer/footer.tsx:48]: `Socials` now takes a styled IconButton.

File: explorer/app/components/Links/Item.tsx [hunk 3]
```diff
@@ -32,14 +37,9 @@ export const Item = ({
   return (
     <Box display="flex">
       <Link href={link.url} passHref>
-        <Text
-          variant="text-body-400-2lines"
-          component="a"
-          customColor="primaryAnvil"
-          sx={{ wordBreak: "break-word" }} // TODO(cc) to be resolved with #89 (Update citation component to match refined mocks)
-        >
+        <MLink variant="text-body-400-2lines">
```
Comment (frano-m) [inline explorer/app/components/Links/Item.tsx:40]: TODO - this component should be superseded by `Link` however it is still in use. Refactored `Text` to a Mui `Link` component to achieve correct link styles.

File: explorer/app/components/Project/components/CollaboratingOrganizations/collaboratingOrganizations.tsx [hunk 4]
```diff
@@ -7,7 +7,7 @@ import { SectionDetailsEmpty } from "../Section/components/SectionDetailsEmpty/s
 import { Section } from "../Section/section";
 
 // Styles
-import { Sup } from "../../../Citations/Citations.styles";
+import { Sup } from "../Sup/Sup.styles";
```
Comment (frano-m) [inline explorer/app/components/Project/components/CollaboratingOrganizations/collaboratingOrganizations.tsx:10]: `Citations` unused and subsequently deleted. `Sup` style in use and refactored to `Project` folder.

File: explorer/app/theme/definitions.d.ts [hunk 5]
```diff
@@ -1,33 +1,33 @@
...
 declare module "@mui/material/styles/createPalette" {
   interface Palette {
-    ink: CustomColors["ink"];
-    inkLight: CustomColors["inkLight"];
-    primaryAnvil: CustomColors["primaryAnvil"];
-    smoke: CustomColors["smoke"];
-    smokeDark: CustomColors["smokeDark"];
-    smokeLight: CustomColors["smokeLight"];
-    smokeLightest: CustomColors["smokeLightest"];
+    alert: PaletteColor;
```
Comment (frano-m) [inline explorer/app/theme/definitions.d.ts:14]: Refactored to reflect similar shape to Mui Palette.

File: explorer/pages/_app.tsx [hunk 6]
```diff
@@ -1,11 +1,13 @@
 // Core dependencies
-import { theme } from "app/theme";
 import { ThemeProvider as Emotion10ThemeProvider } from "@emotion/react";
 import { CssBaseline } from "@mui/material";
 import { ThemeProvider } from "@mui/material/styles";
+import type { AppProps } from "next/app";
```
Comment (frano-m) [inline explorer/pages/_app.tsx:5]: Linting.

File: explorer/app/components/Links/Item.tsx (hunk 3 above)
Comment (MillenniumFalconMechanic) [reply explorer/app/components/Links/Item.tsx:40]: Should we add a corresponding `// TODO` in the code here to record the required update beyond this PR?

File: explorer/app/components/Layout/components/Footer/footer.tsx (hunk 2 above)
Comment (MillenniumFalconMechanic) [reply explorer/app/components/Layout/components/Footer/footer.tsx:48]: Can you give me some background on why this was necessary with respect to the palette updates?

**Review by MillenniumFalconMechanic — APPROVED** [review]
LGTM, thanks F!

File: explorer/app/components/Layout/components/Footer/footer.tsx (hunk 2 above)
Comment (frano-m) [reply explorer/app/components/Layout/components/Footer/footer.tsx:48]: IconButton prop colour expects a top level palette colour like "primary" or "smoke" or "success" or "info". It then takes the defined `main` value for the palette e.g. `success.main`. The social icons are coloured "smoke.light" and so we are unable to specify this colour (using `smoke` would return `smoke.main`). Unfortunately, IconButton does not have the `variant` prop (with which I could then create a special variant and modify the colour in theme). Creating a styled button was the next best "workaround" in this particular instance.

File: explorer/app/components/Links/Item.tsx (hunk 3 above)
Comment (frano-m) [reply explorer/app/components/Links/Item.tsx:40]: Sure! There are a number of components that could become obsolete. I'll make this one up.

**Pull request merged**